File: admin-server/src/main/kotlin/com/systec/umeet/admin/service/SysUserGroupPkService.java
package com.systec.umeet.admin.service;

import com.github.pagehelper.PageHelper;
import com.systec.umeet.admin.dto.UserGroupDto;
import com.systec.umeet.admin.mapper.SysUserGroupPkMapper;
import com.systec.umeet.admin.mapper.SysUserMapper;
import com.systec.umeet.admin.model.SysUserGroupPk;
import com.systec.umeet.common.vo.ResponsePage;
import org.apache.commons.lang3.StringUtils;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

/**
 * @description:
 * @author: super.wu
 * @date: Created in 2018/5/9 0009
 * @modified By:
 * @version: 1.0
 **/
@Service
@Transactional(rollbackFor = Exception.class)
public class SysUserGroupPkService extends BaseService<SysUserGroupPkMapper, SysUserGroupPk> {

    @Autowired
    private SysUserMapper sysUserMapper;

    public void setSysUserMapper(SysUserMapper sysUserMapper) {
        this.sysUserMapper = sysUserMapper;
    }

    public ResponsePage loadGroupNoRelationUsers(UserGroupDto dto){
        PageHelper.startPage(dto.getPage(), dto.getSize());
        Map<String, Object> map = getQueryParams(dto);
        return new ResponsePage<>(sysUserMapper.selectGroupNoUsers(map));
    }

    public ResponsePage loadGroupUsers(UserGroupDto dto){
        PageHelper.startPage(dto.getPage(), dto.getSize());
        Map<String, Object> map = getQueryParams(dto);
        return new ResponsePage<>(sysUserMapper.selectGroupUsers(map));
    }

    private Map<String, Object> getQueryParams(UserGroupDto dto) {
        return new HashMap<String, Object>(1){{
            put("groupId", dto.getGroupId());
            if (StringUtils.isNotBlank(dto.getName())) {
                put("name", dto.getName());
            }
        }};
    }

    public Integer batchInsert(List<SysUserGroupPk> list) {
        return this.mapper.insertGroupUsers(list);
    }

    public Integer batchDelete(int groupId, Long [] userIds) {
        Map<String, Object> map = new HashMap<String, Object>(2){{
            put("groupId", groupId);
            put("userIds", userIds);
        }};
        return this.mapper.deleteGroupUsers(map);
    }

}

File: umeet-admin-server/src/main/kotlin/com/systec/umeet/service/SysUserGroupPkService.kt
package com.systec.umeet.service
import com.github.pagehelper.PageHelper
import com.github.pagehelper.PageInfo
import com.systec.umeet.model.SysUser
import com.systec.umeet.base.BaseServiceImpl
import com.systec.umeet.contants.ResponsePage
import com.systec.umeet.dao.SysUserDao
import com.systec.umeet.dao.SysUserGroupPkDao
import com.systec.umeet.model.SysUserGroup
import com.systec.umeet.model.SysUserGroupPk
import org.apache.commons.lang3.StringUtils
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import tk.mybatis.mapper.common.Mapper
import java.util.*

@Service
class SysUserGroupPkService : BaseServiceImpl<SysUserGroupPk> (){

    @Autowired
    var sysUserGroupPkDao : SysUserGroupPkDao? = null

    @Autowired
    var sysUserDao : SysUserDao ?= null

    override fun getMapper(): Mapper<SysUserGroupPk> {
        return sysUserGroupPkDao!!
    }

    /**
     * 根据条件分页查询用户
     */
    fun getPage(record: SysUserGroupPk): ResponsePage<SysUserGroupPk> {
        return super.selectPage(record.page , record.size ,record)
    }

    fun loadGroupNoRelationUsers(dto: SysUserGroupPk): ResponsePage<*> {
        PageHelper.startPage<Any>(dto.page, dto.size)
        val map = getQueryParams(dto)
        return ResponsePage<SysUser>(sysUserDao!!.selectGroupNoUsers(map))
    }

    fun loadGroupUsers(dto: SysUserGroupPk): ResponsePage<*> {
        PageHelper.startPage<Any>(dto.page, dto.size)
        val map = getQueryParams(dto)
        return ResponsePage<SysUser>(sysUserDao!!.selectGroupUsers(map))
    }

    private fun getQueryParams(dto: SysUserGroupPk): Map<String, Any> {
        return object : HashMap<String, Any>(1) {
            init {
                put("groupId", dto.groupId!!)
                if (StringUtils.isNotBlank(dto.name)) {
                    put("name", dto.name!!)
                }
            }
        }
    }

    fun batchInsert(list: List<SysUserGroupPk>): Int? {
        return sysUserGroupPkDao!!.insertGroupUsers(list)
    }

    fun batchDelete(groupId: Int, userIds: Array<Long>): Int? {
        val map = object : HashMap<String, Any>(2) {
            init {
                put("groupId", groupId)
                put("userIds", userIds)
            }
        }
        return sysUserGroupPkDao!!.deleteGroupUsers(map)
    }
}
